### PR TITLE
topic 0

### PR DESCRIPTION
**Pull request by jordanholtz**
keys 2 and 3. Los uml que estan ahi son simplemente ilustrativos, no
representan el diseño propuesto strictamente hablando, pero si una idea
de lo que ocurre.

### CODE REVIEW

File: src/key2/BrickBuilder.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+public class BrickBuilder extends HouseBuilder {
+
+	@Override
+	public House createHouse() {
+		return new BrickHouse();
+	}
+
+	@Override
+	public Floor createFloor() {
+		// TODO Auto-generated method stub
```
Comment (lucast-globant) [inline src/key2/BrickBuilder.java:12]: Is there any implementation missing here? If not please remove TODO comment.

File: src/key2/BrickFloor.java [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package key2;
+
+public class BrickFloor implements Floor {
+
+	@Override
+	public String getRepresentation() {
+		return "Finished building wood floor";
```
Comment (lucast-globant) [inline src/key2/BrickFloor.java:7]: what is the reason of having this method?

File: src/key2/HouseDirector.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+		
+		return house;
+	}
+	
+	/**
+	 * @param str say something
+	 */
+	public void log(String str){
+		System.out.println(str);
+	}
```
Comment (lucast-globant) [inline src/key2/HouseDirector.java:30]: There was no need for loggin. You can save some time next time just focusing on minimum requeriments.

File: src/key3/HospitalFactory.java [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+	      }
+	      case MUNICIPAL:{
+	           return new Municipal();
+	      }
+	      default:
+	           System.out.println("error" );
+	           break;
+	      }
+		return null;
+	}*/
```
Comment (lucast-globant) [inline src/key3/HospitalFactory.java:25]: Do not include commented code on pull requests. Please remove.

File: src/key3/HospitalFactory.java [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+	
+	public Hospital createHospital(Type hospitalType) {
+		HospitalDirector hd=new HospitalDirector();
+		HospitalBuilder hb;
+		 switch ( hospitalType ) {
+	      case PRIVADO:{
```
Comment (lucast-globant) [inline src/key3/HospitalFactory.java:31]: Only english.

**Comment by lucast-globant** [comment]
Good job! Consider this Topic as closed once you attend the code review comments.

File: src/key2/BrickFloor.java (hunk 2 above)
Comment (jordanholtz) [inline src/key2/BrickFloor.java:7]: Its just a represenation of the floor, nothing especial, is for diferentiate from other floors. There i miss with the return, it should return "finishing building brick floor"

File: src/key2/HouseDirector.java (hunk 3 above)
Comment (jordanholtz) [inline src/key2/HouseDirector.java:30]: its just for that, save time :), i hate log with System.out.println XD. I miss too much the simple C printf.

File: src/key2/BrickFloor.java (hunk 2 above)
Comment (lucast-globant) [inline src/key2/BrickFloor.java:7]: getRepresentation implementation and BrickFloor should not refer to any "building" aspect, that's our builder responsability. Please take a look at  the Single Responsability Principle http://en.wikipedia.org/wiki/SOLID_(object-oriented_design). Keep this in mind for next Topic.

**Pull request closed without merging**